platform/linux-generic/odp_ipsec.c

```diff
@@ -58,15 +58,64 @@ int odp_ipsec_capability(odp_ipsec_capability_t *capa)
 }
 
 int odp_ipsec_cipher_capability(odp_cipher_alg_t cipher,
-				odp_crypto_cipher_capability_t capa[], int num)
+				odp_ipsec_cipher_capability_t capa[], int num)
 {
-	return odp_crypto_cipher_capability(cipher, capa, num);
+	int caps = odp_crypto_cipher_capability(cipher, NULL, 0);
+	odp_crypto_cipher_capability_t crypto_capa[caps];
+	uint32_t req_iv_len;
+	int rc, i, out;
+
+	rc = odp_crypto_cipher_capability(cipher, crypto_capa, caps);
```

Comment:
please add check that caps > 0.

Reply:
Rewrote this piece of code to use fixed length arrays.